src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/SubaccountTransactionSupervisor.kt

```diff
@@ -652,13 +654,24 @@ internal class SubaccountTransactionSupervisor(
         val isolatedMarginTransactionCallback = { response: String? ->
             val error = parseTransactionResponse(response)
             if (error == null) {
-                submitTransaction(
-                    transactionType = TransactionType.PlaceOrder,
-                    transactionPayloadString = string,
-                    onSubmitTransaction = onSubmitOrderTransaction,
-                    transactionCallback = orderTransactionCallback,
-                    useTransactionQueue = useTransactionQueue,
-                )
+                // Return submitTransaction after a delay to ensure the transfer is confirmed
+                val timer = helper.ioImplementations.timer ?: CoroutineTimer.instance
+
+                val delayedPlaceOrder = timer.schedule(
```

Comment:
"timer.run()" is easier

`val delayedPlaceOrder = timer.run(delay = 0.25) { ... }`